librosa/core/audio.py

```diff
@@ -1320,7 +1349,14 @@ def clicks(
     return click_signal
 
 
-def tone(frequency, *, sr=22050, length=None, duration=None, phi=None):
+def tone(
+    frequency: _FloatLike_co,
```

Comment:
True... I went with FloatLike when the tests were calling it with a numpy scalar value (to make the tests pass the typecheck)... This is a real headache with numpy typing because `np.floating` is accept "almost anywhere" float is accepted, but is not exactly the same. In theory I could replace all instances of `float` with `_FloatLike_co` in parameter types, but not sure that is advisable...